core/src/test/java/io/jenkins/pluginhealth/scoring/probes/DefaultBranchBuildStatusProbeTest.java

```diff
@@ -0,0 +1,171 @@
+
+import java.io.IOException;
+import java.time.ZonedDateTime;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+
+import io.jenkins.pluginhealth.scoring.model.Plugin;
+import io.jenkins.pluginhealth.scoring.model.ProbeResult;
+import io.jenkins.pluginhealth.scoring.model.updatecenter.UpdateCenter;
+
+import hudson.util.VersionNumber;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.kohsuke.github.GHCheckRun;
+import org.kohsuke.github.GHCommit;
+import org.kohsuke.github.GHRepository;
+import org.kohsuke.github.GitHub;
+import org.kohsuke.github.PagedIterable;
+import org.kohsuke.github.PagedIterator;
+
+class DefaultBranchBuildStatusProbeTest extends AbstractProbeTest<DefaultBranchBuildStatusProbe> {
+
+    private DefaultBranchBuildStatusProbe probe;
+
+    @Override
+    DefaultBranchBuildStatusProbe getSpy() {
+        return spy(DefaultBranchBuildStatusProbe.class);
+    }
+
+    @BeforeEach
+    public void init() {
+
+        probe = getSpy();
+    }
+
+    @Test
+    public void shouldReturnBuildSucessOntheDefaultBranch() throws IOException {
+
+        final String pluginName = "mailer";
+        final String pluginRepo = "jenkinsci/" + pluginName + "-plugin";
+        final String scmLink = "https://github.com/" + pluginRepo;
+        final String defaultBranch = "main";
+        final GitHub gitHub = mock(GitHub.class);
+        final GHRepository ghRepository = mock(GHRepository.class);
+        final GHCommit ghCommit = mock(GHCommit.class);
+        final GHCheckRun checkRun = mock(GHCheckRun.class);
+        final PagedIterable<GHCheckRun> checkrun = mock(PagedIterable.class);
+        final PagedIterator<GHCheckRun> checkIterator = mock(PagedIterator.class);
+        final Plugin plugin = mock(Plugin.class);
+        final ProbeContext ctx = mock(ProbeContext.class);
+
+        final GitHub gh = mock(GitHub.class);
+
+        when(plugin.getName()).thenReturn(pluginName);
+        when(plugin.getScm()).thenReturn(scmLink);
+        when(ctx.getUpdateCenter())
+                .thenReturn(new UpdateCenter(
+                        Map.of(
+                                pluginName,
+                                new io.jenkins.pluginhealth.scoring.model.updatecenter.Plugin(
+                                        pluginName,
+                                        new VersionNumber("1.0"),
+                                        scmLink,
+                                        ZonedDateTime.now(),
+                                        List.of(),
+                                        0,
+                                        "42",
+                                        defaultBranch)),
+                        Map.of(),
+                        List.of()));
+        when(ctx.getGitHub()).thenReturn(gh);
+        when(ctx.getRepositoryName()).thenReturn(Optional.of(pluginRepo));
+        when(ctx.getGitHub()).thenReturn(gitHub);
```

Comment:
Unless.

Reply:
@alecharp I didn't get you regarding this

Reply:
Yes, I meant useless. you already mock that class before.